velox/functions/sparksql/Map.cpp

```diff
@@ -32,50 +32,27 @@
 namespace facebook::velox::functions::sparksql {
 namespace {
 
-template <TypeKind kind>
-void setKeysResultTyped(
+void setKeysAndValuesResult(
     vector_size_t mapSize,
     std::vector<VectorPtr>& args,
     const VectorPtr& keysResult,
+    const VectorPtr& valuesResult,
     exec::EvalCtx& context,
     const SelectivityVector& rows) {
-  using T = typename KindToFlatVector<kind>::WrapperType;
-  auto flatVector = keysResult->asFlatVector<T>();
-  flatVector->resize(rows.end() * mapSize);
-
   exec::LocalDecodedVector decoded(context);
+
   for (vector_size_t i = 0; i < mapSize; i++) {
     decoded.get()->decode(*args[i * 2], rows);
-    // For efficiency traverse one arg at the time
+    SelectivityVector targetRows(keysResult->size(), false);
+    std::vector<vector_size_t> toSourceRow(keysResult->size());
     rows.applyToSelected([&](vector_size_t row) {
       VELOX_CHECK(!decoded->isNullAt(row), "Cannot use null as map key!");
```

Comment:
This needs be to a USER check and the loop should use context.applyToSelectedNoThrow. Please, add test cases with TRY to verify the behavior.

Reply:
Thanks, updated, it's good to learn this `applyToSelectedNoThrow`.